lib/pages/personal_qr/personal_qr.dart

```diff
@@ -24,6 +25,40 @@ class PersonalQr extends StatefulWidget {
 class PersonalQrController extends State<PersonalQr> {
   final GlobalKey qrKey = GlobalKey();
 
+  /// Shares the QR code as an image file
+  Future<void> shareQrCode(BuildContext context) async {
+    try {
+      final l10n = L10n.of(context)!;
+
+      final imageBytes = await _captureQrImage(context, l10n);
+      if (imageBytes == null) {
+        return;
+      }
+
+      final tempDir = await getTemporaryDirectory();
+      final timestamp = DateTime.now().millisecondsSinceEpoch;
+      final filePath = '${tempDir.path}/qr_code_$timestamp.png';
+      final file = File(filePath);
+      await file.writeAsBytes(imageBytes);
+
+      await Share.shareXFiles(
+        [XFile(file.path)],
+        text: l10n.shareQrCode,
+      );
+
+      // Clean up temp file after a delay to ensure sharing completes
+      Future.delayed(const Duration(seconds: 2), () {
```

Comment:
why need to wait before delete file? How about delete it right after `shareXFiles`?